cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java

```diff
@@ -101,6 +101,10 @@ private static void writeTopologicalIslands(Network network, CgmesExportContext
             for (String tn : island.topologicalNodes) {
                 CgmesExportUtil.writeReference("TopologicalIsland.TopologicalNodes", tn, cimNamespace, writer, context);
             }
+            for (DanglingLine dl : network.getDanglingLines(DanglingLineFilter.ALL)) {
```

Comment:
I don't know if these topological nodes should be added to the topologicalIsland object directly.

Reply:
No, I don't think this is the proper way to do that. Here for each island you are adding all the boundary nodes (since you loop on all dangling line inside the loop). So if there are multiple islands you are attaching the same boundary topological nodes to different islands, which doesn't make sense.

Reply:
You are absolutely right, this is wrong. I tried to add the topological nodes of the boundaries to the topological island but it is also not working because we do not write the topological islands that have no reference terminal. I am going to need help to move forward.